python-modules/cis_identity_vault/cis_identity_vault/models/user.py

```diff
@@ -350,6 +351,17 @@ def _get_segment(self, segment=0, total_segments=10, connection_method=None, act
                 ExclusiveStartKey=response["LastEvaluatedKey"],
             )
             users.extend(response["Items"])
+
+        return users
+
+    async def _resultifier(self, pool_size, connection_method, active):
+        users = []
+        segment_result = None
+        for x in range(0, pool_size):
+            segment_result = await self._get_segment(
```

Comment:
is this really faster, if we await on the longest operation?

Reply:
Yes.

Reply:
nope! ;-) https://github.com/mozilla-iam/cis/pull/463